daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -184,6 +185,38 @@ func (l *logStream) Close() error {
 
 // create creates a log stream for the instance of the awslogs logging driver
 func (l *logStream) create() error {
+
+	// Create log group first
+	{
```

Comment:
@samuelkarp I enclosed the code in a new block level because I wanted to reuse the code below (that creates streams) without changing variable names - I'm not a Go developer, this was the easiest and errorless way for me.
I had a feeling that Docker's awslogs driver doesn't create groups intentionally, but I needed this feature for a real use case, where I generate group names dynamically, and not expecting as much as 500 groups to be generated. Anyway, Amazon's own awslogs agent does exactly that - creates new log group if it doesn't exist. I think it is better for a container to fail to run when group limit is reached rather than fail to run if the group doesn't exist. Or at least let the user decide what is better for him by defining some extra log driver option.